numpy/lib/tests/test_function_base.py

```diff
@@ -668,7 +668,7 @@ def test_ufunc(self):
         args = np.array([0, 0.5*np.pi, np.pi, 1.5*np.pi, 2*np.pi])
         r1 = f(args)
         r2 = np.cos(args)
-        assert_array_equal(r1, r2)
+        assert_array_almost_equal(r1, r2)
```

Comment:
Just a small follow up on this one, I think this is actually the [intel bug](https://randomascii.wordpress.com/2014/10/09/intel-underestimates-error-bounds-by-1-3-quintillion/) in fsin and fcos.
In the coments for that article there is some discussion about how msvc does not exibit this bug (though some other rounding inconsistencies are present). AFAICT, the sin/cos instructions in mingw are still just wrappers around fsin though, so they do exhibit the intel bug. This can be seen at the Python level by doing:

    import numpy as np
    import sympy as sy
    import math as ma
    print sy.Float(np.pi, 100) + sy.Float(ma.sin(np.pi), 100)
    print sy.Float(np.pi, 100) + sy.Float(np.sin(np.pi), 100)

The first number printed is a much better approximation of pi than the second.
This would cause the error in the tests here when compiling with mingw for a python installation compiled with msvc, since the sin and cos functions in the math library come from msvc while the ones in mingw just wrap fsin and fcos.